iib/workers/tasks/build_create_empty_index.py

```diff
@@ -0,0 +1,136 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+import tempfile
+import json
+import re
+
+from iib.workers.api_utils import set_request_state
+from iib.workers.tasks.build import (
+    _add_label_to_index,
+    _build_image,
+    _cleanup,
+    _create_and_push_manifest_list,
+    _get_index_database,
+    _opm_index_rm,
+    _push_image,
+    _serve_index_registry,
+    _update_index_image_build_state,
+    _update_index_image_pull_spec,
+)
+from iib.workers.tasks.celery import app
+from iib.workers.tasks.utils import (
+    request_logger,
+    prepare_request_for_build,
+    run_cmd,
+    RequestConfigCreateIndexImage,
+)
+
+__all__ = ['handle_create_empty_index_request']
+
+log = logging.getLogger(__name__)
+
+
+def _get_present_operators(from_index, base_dir):
+    """
+    Get a list of operators already present in the index image.
+
+    :param str from_index: index image to inspect.
+    :param str base_dir: base directory to create temporary files in.
+    :return: list of unique present operators as provided by the grpc query
+    :rtype: list
+    :raises IIBError: if any of the commands fail.
+    """
+    db_path = _get_index_database(from_index, base_dir)
```

Comment:
Lines 43 to 52 seem to be duplicated in the merge-index-image endpoint. Could you please create a helper function that can be used by both the functions?

Reply:
It looks similar but the grpcurl is different - this one uses 'api.Registry/ListPackages' and merge-index-image and add-bundles uses 'api.Registry/ListBundles'

Reply:
this what you are suggesting is a huge refactoring and moving functions from build.py to utils.py or creating a new file.I am not sure it worth having it.

Reply:
done